azurerm/internal/services/kusto/resource_arm_kusto_eventgrid_data_connection.go

```diff
@@ -0,0 +1,284 @@
+package kusto
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/kusto/mgmt/2019-05-15/kusto"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmKustoEventGridDataConnection() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmKustoEventGridDataConnectionCreateUpdate,
+		Read:   resourceArmKustoEventGridDataConnectionRead,
+		Update: resourceArmKustoEventGridDataConnectionCreateUpdate,
+		Delete: resourceArmKustoEventGridDataConnectionDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
```

Comment:
The timeouts do not make sense. Read timeout should be 1 min. All others should be 30 mins

Reply:
These timeouts chosen as a default across all of our resources. They may seem larger than they should be but it's just an extra precaution in case anything wonky happens on Azure's side that causes these calls to take longer than usual.

Reply:
Ok, they are pretty high though. Read operations receive timeout from Azure ARM API after 20 seconds so a client timeout of 1 min seems reasonable.